src/state.rs

```diff
@@ -250,3 +261,49 @@ impl Stack {
         self.stack[index] = value;
     }
 }
+
+impl Default for Heap {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Heap {
+    pub fn new() -> Self {
+        Self { heap: vec![] }
+    }
+    // Returns how many ergs the expand costs
+    pub fn expand_memory(&mut self, address: u32) -> u32 {
+        if address >= self.heap.len() as u32 {
+            let old_size = self.heap.len() as u32;
+            self.heap.resize(address as usize + 1, 0);
+            return MEMORY_GROWTH_ERGS_PER_BYTE * (address - old_size + 1);
+        }
+        0
+    }
+
+    pub fn store(&mut self, address: u32, value: U256) {
+        for i in 0..32 {
+            self.heap[address as usize + i] = value.byte(31 - i);
+        }
+    }
+
+    pub fn read(&mut self, address: u32) -> U256 {
+        let mut result = U256::zero();
+        for i in 0..32 {
```

Comment:
Here you can also use to_big_endian, but I think that would be overkill here, and what you're doing is more clear, so I would leave it as is.

Reply:
Perfect, I leave it as is